File: app/src/main/java/edu/ycp/cs482/iorc/Fragments/CharacterPanels/EquipmentFragment.java
package edu.ycp.cs482.iorc.Fragments.CharacterPanels;


import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import java.util.HashMap;

import edu.ycp.cs482.iorc.Activities.ItemListActivity;
import edu.ycp.cs482.iorc.R;


/**
 * A simple {@link Fragment} subclass.
 */
public class EquipmentFragment extends Fragment {


    private static final String CREATION_DATA = "CREATION_DATA";
    private HashMap<String, String> creationData;

    public EquipmentFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.fragment_equipment, container, false);

        String[] ownedItems = {"Torch", "Bedroll", "Rope"};

        ListView listView = view.findViewById(R.id.equipmentList);

        ArrayAdapter<String> listViewAdapter = new ArrayAdapter<>(getActivity(),
                android.R.layout.simple_list_item_1, ownedItems);

        listView.setAdapter(listViewAdapter);


        //creationData = (HashMap<String, String>) extra.getSerializable(CREATION_DATA);
        FloatingActionButton fab = view.findViewById(R.id.item_button);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent itemIntent = new Intent(getActivity(), ItemListActivity.class);
                itemIntent.putExtra(CREATION_DATA, creationData);
                startActivity(itemIntent);
            }
        });

        // Inflate the layout for this fragment
        return view;
    }

}

File: app/src/main/java/edu/ycp/cs482/iorc/Fragments/CharacterPanels/EquipmentFragment.kt
package edu.ycp.cs482.iorc.Fragments.CharacterPanels

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.GridLayoutManager
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.google.gson.Gson
import edu.ycp.cs482.iorc.CharacterVersionQuery

import edu.ycp.cs482.iorc.R
import edu.ycp.cs482.iorc.Fragments.CharacterPanels.dummy.DummyContent
import edu.ycp.cs482.iorc.Fragments.CharacterPanels.dummy.DummyContent.DummyItem
import edu.ycp.cs482.iorc.ViewAdapters.MyItemRecyclerViewAdapter
import edu.ycp.cs482.iorc.fragment.ItemData

/**
 * A fragment representing a list of Items.
 *
 *
 * Activities containing this fragment MUST implement the [OnListFragmentInteractionListener]
 * interface.
 */
/**
 * Mandatory empty constructor for the fragment manager to instantiate the
 * fragment (e.g. upon screen orientation changes).
 */
class EquipmentFragment : Fragment() {
    private var mListener: OnListFragmentInteractionListener? = null
    private var mItemsList: List<ItemData>? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (arguments != null) {
            val charMap = arguments.getSerializable(ARG_INV_DATA) as java.util.HashMap<String, String>
        }
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater!!.inflate(R.layout.item_list, container, false)

        // Set the adapter
        if (view is RecyclerView) {
            val context = view.getContext()
            //TODO: put data into view adapter
            view.adapter = MyItemRecyclerViewAdapter(mItemsList, mListener)
        }
        return view
    }


    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context is OnListFragmentInteractionListener) {
            mListener = context
        } else {
            throw RuntimeException(context!!.toString() + " must implement OnListFragmentInteractionListener")
        }
    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    /**
     * This interface must be implemented by activities that contain this
     * fragment to allow an interaction in this fragment to be communicated
     * to the activity and potentially other fragments contained in that
     * activity.
     *
     *
     * See the Android Training lesson [Communicating with Other Fragments](http://developer.android.com/training/basics/fragments/communicating.html) for more information.
     */
    interface OnListFragmentInteractionListener {
        fun onListFragmentInteraction(item: ItemData)
    }

    companion object {

        private val ARG_INV_DATA = "ITEM_INV_DATA"

        fun newInstance(invMap: HashMap<String, String>): EquipmentFragment {
            val fragment = EquipmentFragment()
            val args = Bundle()
            args.putSerializable(ARG_INV_DATA, invMap)
            fragment.arguments = args
            return fragment
        }
    }
}
